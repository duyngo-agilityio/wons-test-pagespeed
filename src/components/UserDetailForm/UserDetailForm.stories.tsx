import type { Meta, StoryObj } from '@storybook/react';
import { fn } from '@storybook/test';

import { MOCK_USER_DETAIL } from '@/mocks';

// Components
import UserDetailForm from '.';

const meta: Meta<typeof UserDetailForm> = {
  title: 'Components/UserDetailForm',
  component: UserDetailForm,
  parameters: {
    layout: 'centered',
    controls: { expanded: true },
  },
  decorators: [
    (Story) => (
      <div className="w-[600px]">
        <Story />
      </div>
    ),
  ],
};

export default meta;
type Story = StoryObj<typeof UserDetailForm>;

export const Default: Story = {
  args: {
    user: MOCK_USER_DETAIL,
    onAvatarChange: fn(),
    onSubmit: fn(),
    onCancel: fn(),
  },
};
