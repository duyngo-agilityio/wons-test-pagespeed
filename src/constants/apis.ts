export const API_BASE_URL = process.env.NEXT_API_BASE_URL || '';

export const API_PATH = {
  SIGN_UP: '/auth/local/register',
  USERS: '/users',
};
