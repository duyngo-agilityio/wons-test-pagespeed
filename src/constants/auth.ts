// Types
import { TRole } from '@/types';

export const AUTH_METHODS = {
  CREDENTIALS: 'credentials',
};

export const ERROR_TYPES = {
  CREDENTIALS_SIGN_IN: 'CallbackRouteError',
};

export const ROLES: TRole[] = [
  {
    id: 3,
    name: 'Admin',
  },
  {
    id: 4,
    name: 'User',
  },
];
