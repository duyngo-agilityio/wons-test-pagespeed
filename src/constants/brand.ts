export const BRANDS = [
  { key: 'apple', label: 'Apple' },
  { key: 'samsung', label: 'Samsung' },
  { key: 'huawei', label: 'Huawei' },
  { key: 'xioami', label: 'Xioami' },
  { key: 'oppo', label: 'Oppo' },
  { key: 'google', label: 'Google' },
];
