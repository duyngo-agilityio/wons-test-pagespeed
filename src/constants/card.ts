export enum LevelCard {
  LOW = 'low',
  MEDIUM = 'medium',
  HIGH = 'high',
}
