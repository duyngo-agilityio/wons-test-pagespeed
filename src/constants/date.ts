export const DEFAULT_RANGE_VALUE_PICKER = 7;
