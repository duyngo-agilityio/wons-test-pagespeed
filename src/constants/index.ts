export * from './apis';
export * from './auth';
export * from './date';
export * from './fonts';
export * from './images';
export * from './messages';
export * from './params';
export * from './regex';
export * from './routes';
export * from './sidebar';
export * from './themes';
export * from './typography';
export * from './messages';
export * from './regex';
export * from './apis';
export * from './auth';
export * from './invoices';
export * from './select';
export * from './pagination';
