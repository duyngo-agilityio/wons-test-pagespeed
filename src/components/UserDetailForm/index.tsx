'use client';
import { memo, useMemo } from 'react';
import isEqual from 'react-fast-compare';

// Libraries
import { Controller, useForm } from 'react-hook-form';
import { z } from 'zod';
import { zodResolver } from '@hookform/resolvers/zod';

// Utils
import {
  clearErrorOnChange,
  isEnableSubmitButton,
  getDirtyState,
} from '@/utils';

// Components
import { Input, AvatarUpload, Button } from '@/components';

// Constants
import { ERROR_MESSAGES } from '@/constants';

// Types
import { IUserFormData } from '@/types';

interface UserDetailFormProps {
  user: IUserFormData;
  onAvatarChange: (file: File) => void;
  onSubmit: (formData: IUserFormData) => Promise<void>;
  onCancel: () => void;
}

const UserDetailForm = ({
  user,
  onAvatarChange,
  onSubmit,
  onCancel,
}: UserDetailFormProps) => {
  // Zod schema for validation
  const userDetailFormSchema = z.object({
    avatar: z.string().nonempty(ERROR_MESSAGES.FIELD_REQUIRED),
    username: z.string().nonempty(ERROR_MESSAGES.FIELD_REQUIRED),
    role: z.string().nonempty(ERROR_MESSAGES.FIELD_REQUIRED),
    fullName: z.string().nonempty(ERROR_MESSAGES.FIELD_REQUIRED),
    email: z
      .string()
      .nonempty(ERROR_MESSAGES.FIELD_REQUIRED)
      .email(ERROR_MESSAGES.FIELD_INVALID('Email')),
  });

  const REQUIRED_FIELDS = ['fullName', 'email'];

  // Define config and props for useForm
  const {
    control,
    formState: { dirtyFields, errors, defaultValues, isSubmitting, isValid },
    clearErrors,
    handleSubmit,
    watch,
  } = useForm<IUserFormData>({
    resolver: zodResolver(userDetailFormSchema),
    mode: 'onBlur',
    reValidateMode: 'onBlur',
    defaultValues: user,
  });

  const dirtyItems = Object.keys(dirtyFields);

  const enableSubmit: boolean = useMemo(
    () => isEnableSubmitButton(REQUIRED_FIELDS, dirtyItems, errors),
    [REQUIRED_FIELDS, dirtyItems, errors],
  );

  const isDisableSubmit = !(
    enableSubmit || !getDirtyState(defaultValues ?? {}, watch())
  );

  const handleFormSubmit = async (formData: IUserFormData) => {
    await onSubmit(formData);

    onCancel();
  };

  return (
    <form
      className="p-[0_30px_0] w-full"
      onSubmit={handleSubmit(handleFormSubmit)}
      data-testid="user-detail-form"
    >
      <div className="m-[30px_0_0] grid grid-cols-1 md:grid-cols-2 gap-[30px]">
        <Controller
          control={control}
          name="avatar"
          rules={{
            required: ERROR_MESSAGES.FIELD_REQUIRED,
          }}
          render={({
            field: { onChange, value, name },
            fieldState: { error },
          }) => (
            <AvatarUpload
              additionalClass="md:justify-self-start"
              value={value ?? ''}
              error={error?.message}
              isDisabled={isSubmitting}
              onChange={(e) => {
                onChange(e);

                // Clear error message on change
                clearErrorOnChange(name, errors, clearErrors);
              }}
              onFileChange={onAvatarChange}
            />
          )}
        />

        <div className="flex gap-[0_15px] items-center justify-end row-start-7 row-end--1 md:row-start-auto  md:row-end-auto">
          <Button
            className="min-w-[93px] h-fit !bg-white font-normal dark:!bg-white text-center !text-blue-500 dark:text-white/70 border-[1px] border-[rgba(58, 54, 219, 0.1)] py-[10px] !rounded-[10px] font-DM-Sans text-[15px] font-normal leading-normal"
            type="button"
            onClick={onCancel}
            isDisabled={isSubmitting}
          >
            Cancel
          </Button>

          <Button
            className="min-w-[93px] h-fit text-[15px] font-medium py-[10px] px-[25px]"
            type="submit"
            isLoading={isSubmitting}
            color="primary"
            isDisabled={isDisableSubmit || isSubmitting || !isValid}
          >
            Save
          </Button>
        </div>

        <Controller
          name="username"
          control={control}
          render={({ field: { value } }) => (
            <Input isDisabled label="User Name" value={value} />
          )}
        />

        <Controller
          name="role"
          control={control}
          render={({ field: { value } }) => (
            <Input isDisabled label="Role" value={value} />
          )}
        />

        <Controller
          name="fullName"
          control={control}
          render={({
            field: { onChange, name, ...rest },
            fieldState: { error },
          }) => (
            <Input
              classNames={{
                inputWrapper: 'bg-gray-200/30',
              }}
              label="Full Name"
              type="text"
              isInvalid={!!error}
              errorMessage={error?.message}
              isDisabled={isSubmitting}
              onChange={(e) => {
                onChange(e.target.value);

                // Clear error message on change
                clearErrorOnChange(name, errors, clearErrors);
              }}
              {...rest}
            />
          )}
        />

        <Controller
          name="email"
          control={control}
          render={({
            field: { onChange, name, ...rest },
            fieldState: { error },
          }) => (
            <Input
              classNames={{
                inputWrapper: 'bg-gray-200/30',
              }}
              label="Email"
              type="email"
              isInvalid={!!error}
              errorMessage={error?.message}
              isDisabled={isSubmitting}
              onChange={(e) => {
                onChange(e.target.value);

                // Clear error message on change
                clearErrorOnChange(name, errors, clearErrors);
              }}
              {...rest}
            />
          )}
        />
      </div>
    </form>
  );
};

export default memo(UserDetailForm, isEqual);
