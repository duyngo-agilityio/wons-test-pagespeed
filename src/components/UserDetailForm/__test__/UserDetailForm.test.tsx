import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

// Constants
import { IMAGES } from '@/constants';

// Components
import UserDetailForm from '../index';

window.URL.createObjectURL = jest.fn();
const onAvatarChangeMock = jest.fn();
const onSubmitMock = jest.fn();
const onCancelMock = jest.fn();
const mockUser = {
  avatar: IMAGES.AVATAR_DEFAULT,
  username: 'admin',
  fullName: 'Super Admin',
  email: 'admin1@gmail.com',
  role: 'Admin',
};

describe('UserDetail', () => {
  const renderUserDetailForm = () =>
    render(
      <UserDetailForm
        user={mockUser}
        onAvatarChange={onAvatarChangeMock}
        onSubmit={onSubmitMock}
        onCancel={onCancelMock}
      />,
    );

  beforeEach(() => {
    jest.clearAllMocks();
    renderUserDetailForm();
  });

  test('render successful', () => {
    const { container } = renderUserDetailForm();

    // Check if the container is in the document
    expect(container).toBeInTheDocument();

    // Match the snapshot
    expect(container).toMatchSnapshot();
  });

  it('submits the form with data', async () => {
    const nameInput = screen.getByRole('textbox', {
      name: /full name full name/i,
    });

    fireEvent.change(nameInput, { target: { value: 'My Name' } });

    const emailInput = screen.getByRole('textbox', { name: /email email/i });

    fireEvent.change(emailInput, { target: { value: 'example@gmail.com' } });

    const buttonSave = screen.getByRole('button', { name: /save/i });

    await waitFor(() => {
      expect(buttonSave).toBeEnabled();
    });

    // Simulate clicking the submit button
    fireEvent.click(buttonSave);

    // Wait for the onSubmitMock to be called
    await waitFor(() => {
      expect(onSubmitMock).toHaveBeenCalled();
    });
  });

  it('is upload avatar successful', async () => {
    const avatarInput = screen.getByTestId('avatar-upload');

    // Simulate uploading a file
    const file = new File(['image'], 'test.png', { type: 'image/png' });

    await userEvent.upload(avatarInput, file);

    const avatarPreview = screen.getByTestId('avatar-preview'); // Assuming alt text for the preview image is set

    expect(avatarPreview).toBeInTheDocument();
  });

  it('is the save button disabled', async () => {
    const nameInput = screen.getByRole('textbox', {
      name: /full name full name/i,
    });

    // Simulate clear input field
    fireEvent.change(nameInput, { target: { value: '' } });

    const buttonSave = screen.getByRole('button', { name: /save/i });

    // Wait for the button submit to be disabled
    await waitFor(() => {
      expect(buttonSave).toBeDisabled();
    });
  });
});
