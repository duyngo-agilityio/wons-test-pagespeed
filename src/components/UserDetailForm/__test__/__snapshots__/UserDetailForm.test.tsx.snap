// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetail render successful 1`] = `
<div>
  <form
    class="p-[0_30px_0] w-full"
    data-testid="user-detail-form"
  >
    <div
      class="m-[30px_0_0] grid grid-cols-1 md:grid-cols-2 gap-[30px]"
    >
      <div
        class="flex flex-col justify-center items-center md:justify-self-start"
      >
        <label
          class="cursor-pointer hover:scale-110 transition-transform"
          for="file"
        >
          <div
            class="rounded-full w-32 h-32 bg-gray-50 dark:bg-gray-600 flex justify-center items-center"
          >
            <img
              alt="Avatar"
              class="rounded-full object-cover w-full h-full"
              data-nimg="1"
              data-testid="avatar-preview"
              decoding="async"
              fetchpriority="high"
              height="128"
              src="http://localhost/_next/image?url=https%3A%2F%2Ft3.ftcdn.net%2Fjpg%2F01%2F26%2F91%2F78%2F360_F_126917812_XlWgkaV9f81Hde4wvmvJWM3huJRvy5EM.webp&w=256&q=75"
              srcset="/_next/image?url=https%3A%2F%2Ft3.ftcdn.net%2Fjpg%2F01%2F26%2F91%2F78%2F360_F_126917812_XlWgkaV9f81Hde4wvmvJWM3huJRvy5EM.webp&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Ft3.ftcdn.net%2Fjpg%2F01%2F26%2F91%2F78%2F360_F_126917812_XlWgkaV9f81Hde4wvmvJWM3huJRvy5EM.webp&w=256&q=75 2x"
              style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
              width="128"
            />
          </div>
        </label>
        <div
          class="group flex-col data-[hidden=true]:hidden w-full group relative justify-end data-[has-label=true]:mt-[29px] hidden"
          data-filled="true"
          data-filled-within="true"
          data-invalid="true"
          data-slot="base"
        >
          <div
            class="h-full flex flex-col"
            data-slot="main-wrapper"
          >
            <div
              class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm gap-3 min-h-10 transition-background motion-reduce:transition-none !duration-150 data-[hover=true]:bg-primary-100 text-primary placeholder:text-primary outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-offset-background !bg-danger-50 data-[hover=true]:!bg-danger-100 group-data-[focus=true]:!bg-danger-50 bg-gray-50 dark:bg-gray-600 hover:data-[hover=true]:bg-gray-200/50 dark:hover:bg-gray-900 focus-within:bg-gray-50 dark:focus-within:bg-gray-600 group-data-[focus=true]:bg-gray-50 dark:group-data-[focus=true]:bg-gray-600 h-12.5 p-3.75 rounded-[10px] group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-blue-500"
              data-slot="input-wrapper"
              style="cursor: text;"
            >
              <div
                class="inline-flex w-full items-center h-full box-border"
                data-slot="inner-wrapper"
              >
                <input
                  accept="image/*"
                  aria-invalid="true"
                  aria-label="Upload Avatar"
                  class="w-full font-normal !outline-none focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text !placeholder:text-danger !text-danger text-blue-800/70 dark:text-white/70 placeholder:text-blue-800/40 dark:placeholder:text-white/40 bg-transparent text-md"
                  data-slot="input"
                  data-testid="avatar-upload"
                  id="file"
                  placeholder=" "
                  type="file"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex gap-[0_15px] items-center justify-end row-start-7 row-end--1 md:row-start-auto  md:row-end-auto"
      >
        <button
          class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 gap-2 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover hover:bg-gray-200/50 dark:hover:bg-gray-900 px-5 min-w-[93px] h-fit bg-white dark:bg-white text-center text-blue-500 dark:text-white/70 border-[1px] border-[rgba(58, 54, 219, 0.1)] py-[10px] rounded-[10px] font-DM-Sans text-[15px] font-normal leading-normal"
          type="button"
        >
          <div
            class="flex items-center"
          >
            Cancel
          </div>
        </button>
        <button
          class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 gap-2 pointer-events-none [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover bg-blue-500 border-transparent dark:bg-purple-600 text-white hover:bg-blue-100 opacity-50 cursor-not-allowed rounded-[10px] border-[1px] min-w-[93px] h-fit text-[15px] font-medium py-[10px] px-[25px]"
          data-disabled="true"
          disabled=""
          type="submit"
        >
          <div
            class="flex items-center"
          >
            Save
          </div>
        </button>
      </div>
      <div
        class="group flex flex-col data-[hidden=true]:hidden w-full opacity-disabled pointer-events-none group relative justify-end data-[has-label=true]:mt-[29px] is-filled"
        data-disabled="true"
        data-filled="true"
        data-filled-within="true"
        data-has-elements="true"
        data-has-label="true"
        data-has-value="true"
        data-slot="base"
      >
        <div
          class="h-full flex flex-col"
          data-slot="main-wrapper"
        >
          <div
            class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm gap-3 min-h-10 pointer-events-none transition-background motion-reduce:transition-none !duration-150 data-[hover=true]:bg-primary-100 text-primary placeholder:text-primary outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-offset-background bg-gray-50 dark:bg-gray-600 hover:data-[hover=true]:bg-gray-200/50 dark:hover:bg-gray-900 focus-within:bg-gray-50 dark:focus-within:bg-gray-600 group-data-[focus=true]:bg-gray-50 dark:group-data-[focus=true]:bg-gray-600 h-12.5 p-3.75 rounded-[10px] group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-blue-500 is-filled"
            data-slot="input-wrapper"
            style="cursor: text;"
          >
            <label
              class="absolute origin-top-left rtl:origin-top-right subpixel-antialiased block pointer-events-none will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity] text-primary group-data-[filled-within=true]:pointer-events-auto pb-0 z-20 -translate-y-1/2 group-data-[filled-within=true]:left-0 left-3 rtl:left-auto rtl:right-3 group-data-[filled-within=true]:-translate-y-[calc(100%_+_theme(fontSize.small)/2_+_20px)] pe-2 max-w-full text-ellipsis overflow-hidden group-data-[filled-within=true]:text-blue-800 dark:group-data-[filled-within=true]:text-white font-medium text-xl top-5"
              data-slot="label"
              for="react-aria-:ru:"
              id="react-aria-:rv:"
            >
              User Name
            </label>
            <div
              class="inline-flex w-full items-center h-full box-border"
              data-slot="inner-wrapper"
            >
              <input
                aria-label="User Name"
                aria-labelledby="react-aria-:ru: react-aria-:rv:"
                class="w-full font-normal !outline-none focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-blue-800/70 dark:text-white/70 placeholder:text-blue-800/40 dark:placeholder:text-white/40 bg-transparent text-md is-filled"
                data-filled="true"
                data-filled-within="true"
                data-slot="input"
                disabled=""
                id="react-aria-:ru:"
                placeholder=" "
                type="text"
                value="admin"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="group flex flex-col data-[hidden=true]:hidden w-full opacity-disabled pointer-events-none group relative justify-end data-[has-label=true]:mt-[29px] is-filled"
        data-disabled="true"
        data-filled="true"
        data-filled-within="true"
        data-has-elements="true"
        data-has-label="true"
        data-has-value="true"
        data-slot="base"
      >
        <div
          class="h-full flex flex-col"
          data-slot="main-wrapper"
        >
          <div
            class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm gap-3 min-h-10 pointer-events-none transition-background motion-reduce:transition-none !duration-150 data-[hover=true]:bg-primary-100 text-primary placeholder:text-primary outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-offset-background bg-gray-50 dark:bg-gray-600 hover:data-[hover=true]:bg-gray-200/50 dark:hover:bg-gray-900 focus-within:bg-gray-50 dark:focus-within:bg-gray-600 group-data-[focus=true]:bg-gray-50 dark:group-data-[focus=true]:bg-gray-600 h-12.5 p-3.75 rounded-[10px] group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-blue-500 is-filled"
            data-slot="input-wrapper"
            style="cursor: text;"
          >
            <label
              class="absolute origin-top-left rtl:origin-top-right subpixel-antialiased block pointer-events-none will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity] text-primary group-data-[filled-within=true]:pointer-events-auto pb-0 z-20 -translate-y-1/2 group-data-[filled-within=true]:left-0 left-3 rtl:left-auto rtl:right-3 group-data-[filled-within=true]:-translate-y-[calc(100%_+_theme(fontSize.small)/2_+_20px)] pe-2 max-w-full text-ellipsis overflow-hidden group-data-[filled-within=true]:text-blue-800 dark:group-data-[filled-within=true]:text-white font-medium text-xl top-5"
              data-slot="label"
              for="react-aria-:r13:"
              id="react-aria-:r14:"
            >
              Role
            </label>
            <div
              class="inline-flex w-full items-center h-full box-border"
              data-slot="inner-wrapper"
            >
              <input
                aria-label="Role"
                aria-labelledby="react-aria-:r13: react-aria-:r14:"
                class="w-full font-normal !outline-none focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-blue-800/70 dark:text-white/70 placeholder:text-blue-800/40 dark:placeholder:text-white/40 bg-transparent text-md is-filled"
                data-filled="true"
                data-filled-within="true"
                data-slot="input"
                disabled=""
                id="react-aria-:r13:"
                placeholder=" "
                type="text"
                value="Admin"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="group flex flex-col data-[hidden=true]:hidden w-full group relative justify-end data-[has-label=true]:mt-[29px] is-filled"
        data-filled="true"
        data-filled-within="true"
        data-has-elements="true"
        data-has-label="true"
        data-has-value="true"
        data-slot="base"
      >
        <div
          class="h-full flex flex-col"
          data-slot="main-wrapper"
        >
          <div
            class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm gap-3 min-h-10 transition-background motion-reduce:transition-none !duration-150 data-[hover=true]:bg-primary-100 text-primary placeholder:text-primary outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background dark:bg-gray-600 hover:data-[hover=true]:bg-gray-200/50 dark:hover:bg-gray-900 focus-within:bg-gray-50 dark:focus-within:bg-gray-600 group-data-[focus=true]:bg-gray-50 dark:group-data-[focus=true]:bg-gray-600 h-12.5 p-3.75 rounded-[10px] bg-gray-200/30 is-filled"
            data-slot="input-wrapper"
            style="cursor: text;"
          >
            <label
              class="absolute pointer-events-none origin-top-left rtl:origin-top-right subpixel-antialiased block will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity] text-primary group-data-[filled-within=true]:pointer-events-auto pb-0 z-20 -translate-y-1/2 group-data-[filled-within=true]:left-0 left-3 rtl:left-auto rtl:right-3 group-data-[filled-within=true]:-translate-y-[calc(100%_+_theme(fontSize.small)/2_+_20px)] pe-2 max-w-full text-ellipsis overflow-hidden group-data-[filled-within=true]:text-blue-800 dark:group-data-[filled-within=true]:text-white font-medium text-xl top-5"
              data-slot="label"
              for="react-aria-:r18:"
              id="react-aria-:r19:"
            >
              Full Name
            </label>
            <div
              class="inline-flex w-full items-center h-full box-border"
              data-slot="inner-wrapper"
            >
              <input
                aria-label="Full Name"
                aria-labelledby="react-aria-:r18: react-aria-:r19:"
                class="w-full font-normal !outline-none focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-blue-800/70 dark:text-white/70 placeholder:text-blue-800/40 dark:placeholder:text-white/40 bg-transparent text-md is-filled"
                data-filled="true"
                data-filled-within="true"
                data-slot="input"
                id="react-aria-:r18:"
                placeholder=" "
                type="text"
                value="Super Admin"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="group flex flex-col data-[hidden=true]:hidden w-full group relative justify-end data-[has-label=true]:mt-[29px] is-filled"
        data-filled="true"
        data-filled-within="true"
        data-has-elements="true"
        data-has-label="true"
        data-has-value="true"
        data-slot="base"
      >
        <div
          class="h-full flex flex-col"
          data-slot="main-wrapper"
        >
          <div
            class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm gap-3 min-h-10 transition-background motion-reduce:transition-none !duration-150 data-[hover=true]:bg-primary-100 text-primary placeholder:text-primary outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background dark:bg-gray-600 hover:data-[hover=true]:bg-gray-200/50 dark:hover:bg-gray-900 focus-within:bg-gray-50 dark:focus-within:bg-gray-600 group-data-[focus=true]:bg-gray-50 dark:group-data-[focus=true]:bg-gray-600 h-12.5 p-3.75 rounded-[10px] bg-gray-200/30 is-filled"
            data-slot="input-wrapper"
            style="cursor: text;"
          >
            <label
              class="absolute pointer-events-none origin-top-left rtl:origin-top-right subpixel-antialiased block will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity] text-primary group-data-[filled-within=true]:pointer-events-auto pb-0 z-20 -translate-y-1/2 group-data-[filled-within=true]:left-0 left-3 rtl:left-auto rtl:right-3 group-data-[filled-within=true]:-translate-y-[calc(100%_+_theme(fontSize.small)/2_+_20px)] pe-2 max-w-full text-ellipsis overflow-hidden group-data-[filled-within=true]:text-blue-800 dark:group-data-[filled-within=true]:text-white font-medium text-xl top-5"
              data-slot="label"
              for="react-aria-:r1d:"
              id="react-aria-:r1e:"
            >
              Email
            </label>
            <div
              class="inline-flex w-full items-center h-full box-border"
              data-slot="inner-wrapper"
            >
              <input
                aria-label="Email"
                aria-labelledby="react-aria-:r1d: react-aria-:r1e:"
                class="w-full font-normal !outline-none focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-blue-800/70 dark:text-white/70 placeholder:text-blue-800/40 dark:placeholder:text-white/40 bg-transparent text-md is-filled"
                data-filled="true"
                data-filled-within="true"
                data-slot="input"
                id="react-aria-:r1d:"
                placeholder=" "
                type="email"
                value="admin1@gmail.com"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
